src/mod_roster.erl

```diff
@@ -224,7 +227,11 @@ hooks(Host) ->
      {remove_user, Host, ?MODULE, remove_user, 50},
      {anonymous_purge_hook, Host, ?MODULE, remove_user, 50},
      {roster_get_versioning_feature, Host, ?MODULE, get_versioning_feature, 50},
-     {get_personal_data, Host, ?MODULE, get_personal_data, 50}].
+     {get_personal_data, Host, ?MODULE, get_personal_data, 50},
+     %% because mongoose_c2s_presence is not a mod_, we have to either attach its handler here,
```

Comment:
The current solution is OK as long as the handler needs to be registered iff `mod_roster` is enabled.